.travis.yml

```diff
@@ -11,44 +15,57 @@ services:
   - docker
   - redis-server
 
-stages:
-  - check
-  - build
-  - test
-  - run
-
+# NOTE: The smoke test itself will fail because the Travis redis instance does
+# not have keyspace notifications enabled. But we're interested in exercising
+# the testground logic, not the test plan itself. So it's fine.
 jobs:
   include:
-    - stage: check
+    - stage: lint_build_check_test_run
+      os: linux
       script:
-        - go install github.com/golangci/golangci-lint/cmd/golangci-lint
+        - echo 'travis_fold:start:go.mod.graph' && echo 'go mod graph' && go mod graph && echo 'travis_fold:end:go.mod.graph'
+        - GO111MODULE=on pushd .. && go get github.com/golangci/golangci-lint/cmd/golangci-lint && popd
         - golangci-lint run       # run a bunch of code checkers/linters in parallel
-
-    - stage: build
-      script:
         - go build ./...          # build the project
-
-    - stage: test
-      os: linux
-      script:
-        - go test -v -race ./...  # Run all the tests with the race detector enabled
-
-    - stage: test
-      os: osx
-      script:
         - go test -v -race ./...  # Run all the tests with the race detector enabled
+        - go build .
```

Comment:
```suggestion
```

only build once

Reply:
ah, got your memo now. So I guess we do need to build twice?

Reply:
AFAIK, yes.